### ISSUE TITLE
Improve UX around ephemeral entities and backend entity management

### ISSUE THREAD

**Issue by palourde** [sensu/sensu-go#3922]
As noted by https://github.com/sensu/sensu-go/issues/3919, at the moment agents configured with `--deregister` will not be able to update their config unless they wait for their entities to get deregistered by the keepalive mechanism before restarting.

This is not ideal because it essentially means you would have to wait for the keepalive interval after shutting down your agent, before starting it back.

A potential solution we discussed in Slack would be to handle entities with `deregister: true` in a different manner:

- We wouldn't send entity updates to the agent via the session
- We would actively revert changes made to the stored entity config when the watch action is an update, like [we do for entities switch from AgentClass to ProxyClass](https://github.com/sensu/sensu-go/blob/875e7e74bc2fdcc8f4d179192839e2d96e8c2381/backend/agentd/session.go#L288-L311)

**Comment by portertech** [comment]
I think this functionality is a separate feature/field/flag altogether. De-registration is still a good generic feature, which may be applied w/ backend agent entity management.

**Comment by portertech** [comment]
Perhaps something along the lines of:

- `override_entity: true`
- `manage_entity: true`
...

**Comment by calebhailey** [comment]
Or `force_entity_update: true`?

**Comment by calebhailey** [comment]
Removing the `wontfix` label to avoid auto-closing, but I wonder if this is still relevant given the new agent-managed entity improvements in 6.2? /cc @portertech
